digest.go

```diff
@@ -109,18 +109,11 @@ func (d Digest) Validate() error {
 		return ErrDigestInvalidFormat
```

Comment:
I am only talking about the regexp check, `foobar` would get caught for lacking `:`. And who knows, `!!!` could be a valid encoding for `foobar`. It is better for the algorithm to make that distinction since we are only talking here whether `ErrDigestUnsupported` or `ErrDigestInvalidFormat` is returned, both of which indicate invalid input.